### ISSUE TITLE
Feature request: quick fix to make GEMINI more scalable for power users

### ISSUE THREAD

**Issue by schelhorn** [chapmanb/bcbio-nextgen#966]
There is a quick fix possible for making GEMINI more scalable within bcbio. Right now, if you have CADD enabled, all worker nodes have to sift through (or even copy over, I do not now) the large genome-wide SNP file in the `gemini_data` directory. With a hundred or so nodes accessing that 30GB file in parallel with random IO, most shared file systems will give up quite quickly. Thankfully, it is possible to run GEMINI with the `--annotation_dir` parameter that let's you set the location of the `gemini_data` directory at GEMINI run time. 

My feature request therefore is the following: would it be possible to define the location of the GEMINI data directory in bcbio's `system-config.yaml` and automatically provide that location to all calls to GEMINI within bcbio using `--annotation_dir`? In that manner users (using external tools) could sync over the `gemini_data` to node-local storage during provisioning of cluster nodes, change the path of the `annotation_dir` within `system-config.yaml` (again using external tools), and use that system config during bcbio runs (since a system config can dynamically be supplied to `bcbio_nextgen.py` as parameter). Users would then set the path back to the original bcbio location during bcbio updates so that the updater can find the GEMINI data just fine, so the bcbio installation routine wouldn't have to be changed. That's our strategy for the `genomes` data so far, and it works just fine for limiting network IO for genome references on largish installations.

**Comment by schelhorn** [comment]
Okay, after re-reading I realize it probably isn't a _quick_ fix, but if someone would like to undertake the implementation at some point in time I would appreciate it nevertheless :) I guess so would other people _cough_ X Ten _cough_.

**Comment by chapmanb** [comment]
Sven-Eric;
Thanks as always for the suggestions, and for isolating the problem on your systems. I added support for GEMINI to be able to do this within the resources section of your `bcbio_system.yaml` or within sample YAML files, whatever works best (https://bcbio-nextgen.readthedocs.org/en/latest/contents/configuration.html#resources):
```
    resources:
      gemini:
        options: [--annotation-dir, "/path/to/gemini-dbs"]
```
Hope this works for what you need -- let us know if you run into any issues.

**Comment by schelhorn** [comment]
Excellent - very elegant solution, too. Thanks a lot.
